docs/getting-started/openshift/index.md

```diff
@@ -0,0 +1,212 @@
+
+Create the PKI on the principal:
+
+```
+argocd-agentctl --principal-context <control plane context> pki init
+```
+
+Issue the certificate for the principal's gRPC service:
+The gRPC service is the service the agents will connect to. This service usually needs to be exposed to the outside world, or at least from your other clusters where the agents are running on. Typically, this means the service is exposed by a load balancer or a node port.
+
+```
+argocd-agentctl pki issue principal \
+    --principal-context <control plane context> \
+    --ip <ip addresses of principal> \
+    --dns <dns names of principal>
+```
+
+For the --ip and --dns values, you want to specify all addresses and DNS names that the principal will be reachable at for the agents, for example --ip 5.5.5.5 --dns my-principal.example.com. If there is a load balancer in front of the principal, make sure it will pass through traffic to the principal - otherwise, client certificates might not be used for authentication.
+
+Issue the certificate for the principal's resource proxy:
+The principal's resource proxy needs to be reachable by your Argo CD API server (argocd-server), which usually runs in the same cluster as the principal. Typically, this service should not be reachable outside the cluster and is exposed using a Kubernetes Service.
+
+```
+argocd-agentctl pki issue resource-proxy \
+    --principal-context <control plane context> \
+    --ip <ip addresses of resource proxy> \
+    --dns <dns names of principal>
+```
+
+Deploy principal on hub cluster using argocd-operator/gitops-operator using Argo CD CR given below
+
+```
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: argocd
+spec:
+  controller:
+    enabled: false
+  argoCDAgent:
+    principal:
+      enabled: true
+      allowedNamespaces: 
+        - "*"
+      jwtAllowGenerate: true
+      auth: "mtls:CN=([^,]+)"
+      logLevel: "trace"
+      image: "ghcr.io/argoproj-labs/argocd-agent/argocd-agent:latest"
+  sourceNamespaces:
+    - "agent-managed"
+    - "agent-autonomous"  					
+```
+
+The above CR should create all the necessary resource for Argo CD as well as argocd-agent principal in argocd namespace.
+
+Create argocd-redis secret, because principal looks for it to fetch redis authentication details.
+
+```
+oc create secret generic argocd-redis -n argocd --from-literal=auth="$(oc get secret argocd-redis-initial-password -n argocd -o jsonpath='{.data.admin\.password}' | base64 -d)"
+```
+
+## Setting up agent workload cluster 
+
+### Configure Argo-CD for Agent 
+ 
+Argo CD instance on Agent cluster
+
+Creating Argo CD instance for Workload/spoke cluster.
+```
+  apiVersion: argoproj.io/v1beta1
+  kind: ArgoCD
+  metadata:
+    name: argocd
+  spec:
+    server:
+      enabled: false
```

Comment:
```
  redis:
    enabled: true
```

For managed mode, the agent deployment is looking for the `argocd-redis` secret so I am assuming the redis is required.

```
    state:
      waiting:
        message: secret "argocd-redis" not found
        reason: CreateContainerConfigError
```

So the 

`kubectl create secret generic argocd-redis -n argocd --from-literal=auth="$(kubectl get secret argocd-redis-initial-password -n argocd -o jsonpath='{.data.admin\.password}' | base64 -d)"`

is needed as well afterwards.

Reply:
I think it will be required for both the Modes, I have added it for both the sections, managed and autonomous.
I have updated the command 
```
kubectl create secret generic argocd-redis -n <workload namespace> --from-literal=auth="$(kubectl get secret argocd-redis-initial-password -n <argocd-namespace> -o jsonpath='{.data.admin\.password}' | base64 -d)"
```
`workload-namespace` and `argocd-namespace` are mentioned as agent can be present in a different namespace than argocd.

Also, we will not need to enable redis explicitly, as it's absent in the mentioned argocd CR, it will be created with default values.

Reply:
Right. Thanks.